eng-wasm/src/crypto_wasm.rs

```diff
@@ -4,54 +4,30 @@ use eng_pwasm_abi::types::U256;
 use rand_wasm::Rand;
 
 const SYMMETRIC_KEY_SIZE: usize = 32;
-const IV_SIZE: usize = 96 / 8;
-const PAYLOAD_SIZE: usize = 1024;
 
-pub type IV = [u8; IV_SIZE];
 pub type SymmetricKey = [u8; SYMMETRIC_KEY_SIZE];
 
+fn extra_size_for_cypher() -> usize {
+    unsafe {(external::get_iv_size() + external:: get_tag_size()) as usize }
+}
+
+
 pub fn generate_key() -> SymmetricKey {
     let key_int: U256 = Rand::gen();
     let key = H256::from(key_int);
     key.0
 }
 
 pub fn encrypt(message: &[u8], key: &SymmetricKey) -> Vec<u8> {
-    // TODO: Is this really needed? Dynamically sized buffers don't seem to work.
-    // TODO: Is it possible to estimate the encrypted payload size based on the plaintext message?
-    let length = PAYLOAD_SIZE;
-    let mut payload = Vec::with_capacity(length);
-    for _ in 0..length {
-        payload.push(0);
-    }
+    let length = message.len() + extra_size_for_cypher();
```

Comment:
Are we sure this is also working? If extra_size..() is constant, but message.len() is obviously dynamic, how can we be sure we allocate the exact size of the encrypted message?